src/lavinmq/clustering/client.cr

```diff
@@ -228,11 +228,12 @@ module LavinMQ
             else
               File.delete? File.join(@data_dir, filename)
             end
-          when .positive? # full file is coming
-            Log.debug { "Getting full file #{filename} (#{len} bytes)" }
-            f = @files[filename]
-            f.truncate
+          when .positive? # replace file
+            Log.debug { "Replacing file #{filename} (#{len} bytes)" }
+            f = @files["#{filename}.tmp"]
             IO.copy(lz4, f, len) == len || raise IO::EOFError.new("Full file not received")
+            f.rename f.path[0..-5]
+            @files.delete("#{filename}.tmp")
```

Comment:
I don't know if it's necessary to add the file back to the hash with it's new name, it would still be opened next time it's accessed.

```suggestion
            @files[filename] = @files.delete("#{filename}.tmp")
```

If not adding it back we can probably close it right away instead of letting it happen during GC.

Reply:
> I don't know if it's necessary to add the file back to the hash with it's new name, it would still be opened next time it's accessed.

Yeah, shouldn't be necessary to add it back I think.

> If not adding it back we can probably close it right away instead of letting it happen during GC.

Hmm, yeah. Not sure if I understand how that suggested change accomplishes that though?

Reply:
```suggestion
            @files.delete("#{filename}.tmp").close
```